seam/client.py

```diff
@@ -19,13 +20,21 @@
 
 
 class SeamHttpClient(requests.Session, AbstractSeamHttpClient):
-    def __init__(self, base_url: str, auth_headers: Dict[str, str], **kwargs):
+    def __init__(
+        self,
+        base_url: str,
+        auth_headers: Dict[str, str],
+        retry_config: Optional[Retry] = None,
+        **kwargs
+    ):
         super().__init__(**kwargs)
 
         self.base_url = base_url
         headers = {**auth_headers, **kwargs.get("headers", {}), **SDK_HEADERS}
 
         self.headers.update(headers)
+        if retry_config:
+            self.retries = retry_config
```

Comment:
We should set a default Retry config here if None.

Reply:
What should be the default retry config? Something like this is ok?
```py
DEFAULT_RETRIES = Retry(
    total=2,
    backoff_factor=0.25,
    status_forcelist=[500, 502, 503, 504],
    allowed_methods=["HEAD", "GET", "PUT", "DELETE", "OPTIONS", "TRACE", "POST"],
    raise_on_status=False,
)
```

Reply:
IMO we should not actually allow retry on POST (which I realize means this basically won't do anything). In theory it's safe, but not without proper idempotency key support (which is an eventual goal).

See: https://github.com/seamapi/javascript-http/blob/295dc1707b8e6e1fdf82d53311e0f152ac74a11b/test/seam/connect/retry.test.ts#L26-L30

For the default, we can either respect `Retry`, so just `retries=Retry()`

Or we can define an SDK standard retry strategy and set the defaults to match across all the SDKs. Currenlty, that would mean starting with the axios-retry defaults from the JS SDK and matching them as closely as we can to Rety options. If we do this, we can eventually "upgrade" the default to support POST when it's safe.

There is one other way I can think of to improve the Retry behavior without GET support, and that would be to enable POST rety via codegen if `semanticMethod` in the blueprint was a safe method to retry.

Reply:
>IMO we should not actually allow retry on POST (which I realize means this basically won't do anything). In theory it's safe, but not without proper idempotency key support (which is an eventual goal).

Oh yeah, I agree. Great point! 

> Or we can define an SDK standard retry strategy and set the defaults to match across all the SDKs.

I tried to make something similar to what JS SDK implement with `axios-retry`:
```py
DEFAULT_RETRIES = Retry(
    total=2,
    backoff_factor=0.1,
    status_forcelist=[429] + list(range(500, 600)),
    allowed_methods=["GET", "HEAD", "OPTIONS", "PUT", "DELETE"],
    respect_retry_after_header=True,
    raise_on_status=False,
)
```
But I'm unsure about it, let me know what you think. Or we can stick to `retries=Retry()` for now.

> There is one other way I can think of to improve the Retry behavior without GET support, and that would be to enable POST rety via codegen if semanticMethod in the blueprint was a safe method to retry.

I like this. We can come back to this once blueprint is done and the codegen uses it.